editor/post-title/index.js

```diff
@@ -76,6 +89,26 @@ class PostTitle extends Component {
 		this.setState( { isSelected: false } );
 	}
 
+	setFocused( focused ) {
+		this.setState( { hasFocusWithin: focused } );
+		if ( focused ) {
+			this.props.clearSelectedBlock();
+		}
+	}
+
+	onOuterFocus() {
+		clearTimeout( this.blurTimer );
+		this.setFocused( true );
+	}
+
+	onOuterBlur( e ) {
+		const target = e.currentTarget;
+		clearTimeout( this.blurTimer );
+		this.blurTimer = setTimeout( () => {
```

Comment:
`setTimeout` is a code smell in components and can often lead to difficult-to-maintain components. It's often a signal that we don't have a full understanding of the natural flow of events and deferring is a convenient waiting mechanism. I'm wondering if there is a better way to implement this.

Reply:
Yeah, I have the setTimeout too, however I can't see a current way around it. Maybe you can. I'll explain what I think are the sequence of events

a) you focus inside the post title (fires focus event)
b) you click the copied button
  - fires a blur first on the post title (which would cause unselect)
  - then fires a focus on the copy button (based on the mousedown)
  - then fires a click on the copy button (which is what clipboard library is relying on)
  - this then creates a fake textarea and focuses it, does the copying magic and then removes it from the DOM
  - then I return focus to the textarea, because now the focused element is removed from the DOM. I think you are right, though, that we could use the returnFocus HOC.

The big problem is that blur fires before the next DOM element is focused. This is an issue with most frameworks. Some browsers might supply a relatedTarget on the blur event, but not all of them ... which has typically led to setTimeout solutions abounding like https://medium.com/@jessebeach/dealing-with-focus-and-blur-in-a-composite-widget-in-react-90d3c3b49a9b and https://gist.github.com/pstoica/4323d3e6e37e8a23dd59 and https://stackoverflow.com/questions/11592966/get-the-newly-focussed-element-if-any-from-the-onblur-event/11592974#11592974. 

However, there might be a way. In our previous dealings with these issues, we've often had to resort to a setTimeout. Hopefully, you're aware of a better option.